cmd/kube-proxy/app/server_others.go

```diff
@@ -317,6 +347,76 @@ func newProxyServer(
 	}, nil
 }
 
+func getDetectLocalMode(config *proxyconfigapi.KubeProxyConfiguration) proxyconfigapi.LocalMode {
+	switch config.DetectLocalMode {
+	case proxyconfigapi.LocalModeClusterCIDR:
+		return config.DetectLocalMode
+	default:
+		klog.V(4).Info("Defaulting detect local mode to ", string(proxyconfigapi.LocalModeClusterCIDR))
+		return proxyconfigapi.LocalModeClusterCIDR
+	}
+}
+
+func getIPTablesDetectLocal(mode proxyconfigapi.LocalMode, config *proxyconfigapi.KubeProxyConfiguration, ipt utiliptables.Interface) (detectLocal utiliptables.DetectLocalTraffic, err error) {
+	detectLocal = utiliptables.NewNoOpDetectLocal()
+	switch mode {
+	case proxyconfigapi.LocalModeClusterCIDR:
+		if len(config.ClusterCIDR) != 0 {
+			return iptables.NewDetectLocalByCIDR(config.ClusterCIDR, ipt)
+		}
+	}
+	klog.V(2).Info("Detect local mode: ", string(mode), " , defaulting to no-op detect local")
+	return
+}
+
+func getIPTablesDualStackDetectLocal(mode proxyconfigapi.LocalMode, config *proxyconfigapi.KubeProxyConfiguration, ipt [2]utiliptables.Interface) (detectLocal [2]utiliptables.DetectLocalTraffic, err error) {
+	detectLocal = [2]utiliptables.DetectLocalTraffic{utiliptables.NewNoOpDetectLocal(), utiliptables.NewNoOpDetectLocal()}
```

Comment:
same here, set in the default case?

Reply:
actually it's not the default case. For example, it also applies when user selected LocalModeClusterCIDR, but did not set a ClusterCIDR or gave wrong values for the CIDR tuple. In that case too, we fallback to NoOpDetectLocal. I was thinking default would complicate the flow further?